# **You're A Winner, Baby!**

**Author:** Anton Reyes

## **Introduction**

### **Requirements and Imports**

#### **Imports**

**Basic Libraries**

* `numpy` contains a large collection of mathematical functions
* `pandas` contains functions that are designed for data manipulation and data analysis

In [136]:
import numpy as np
import pandas as pd

**Visualization Libraries**

* `matplotlib.pyplot` contains functions to create interactive plots
* `seaborn` is a library based on matplotlib that allows for data visualization
* `plotly` is an open-source graphing library for Python.

In [137]:
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

**Natural Language Processing Libraries**
* `re` is a module that allows the use of regular expressions

In [138]:
import re

#### **Datasets and Files**

The following `csv` file was used for this project:

- `finalists.csv` contains all the finalists of the Drag Race franchise as well as the placements of each contestant in their seasons. Finalists are as of April 22, 2023

## **Data Collection**

Importing the dataset

In [139]:
dataset = "data/finalists.csv"

df = pd.read_csv(dataset)
df.head()

Rank WRE   Rank Country Season    Code W/R/E          Queen  Z-Score  \
0       NaN    NaN     NaN    NaN     NaN   NaN            NaN      NaN   
1      18.0    8.0      US    AS1  US_AS1     W  Chad Michaels     0.47   
2     131.0  159.0      US    AS1  US_AS1     R          Raven    -2.26   
3      77.0  159.0      US    AS1  US_AS1     E        Jujubee    -2.26   
4       6.0    8.0      US    AS1  US_AS1     E        Shannel     0.47   

    Percent  Score  ...  Low  Safe  High  SWin  Win  LSA    T  Unnamed: 36  \
0       NaN    NaN  ...   -1     0     1   1.5    2  NaN  NaN     Never/--   
1   100.00%    5.0  ...    0     0     1   0.0    3  NaN  NaN           --   
2  -120.00%   -6.0  ...    1     0     1   0.0    0  LSA  2.0           --   
3  -120.00%   -6.0  ...    1     0     1   0.0    0  LSA  2.0           --   
4   100.00%    5.0  ...    0     0     1   0.0    3  NaN  NaN           --   

   Unnamed: 37  Unnamed: 38  
0         Rank         Rank  
1        1,220            1  
2          237          140  
3          237          140  
4        1,220            1  

[5 rows x 39 columns]

## **Description of the Dataset**

Here, we find the shape of the dataset.

In [140]:
df.shape

(161, 39)

By looking at the `info` of the dataframe, we can see that there are `non-null` values. 

In [141]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 161 entries, 0 to 160
Data columns (total 39 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Rank WRE     160 non-null    float64
 1   Rank         160 non-null    float64
 2   Country      160 non-null    object 
 3   Season       160 non-null    object 
 4   Code         160 non-null    object 
 5   W/R/E        160 non-null    object 
 6   Queen        160 non-null    object 
 7   Z-Score      160 non-null    float64
 8   Percent      160 non-null    object 
 9   Score        160 non-null    float64
 10  Episode      160 non-null    float64
 11  1            154 non-null    float64
 12  2            154 non-null    float64
 13  3            156 non-null    float64
 14  4            160 non-null    float64
 15  5            159 non-null    float64
 16  6            142 non-null    float64
 17  7            139 non-null    float64
 18  8            93 non-null     float64
 19  9       

## **Exploratory Data Analysis Part 1**

The following questions are asked to guide the EDA.

1. How many finalists are there per class?
2. How many unique values are there in the `Season` column?
3. How many unique values are there in the `Country` column?
4. What are measures of central tendency in the `Episode` column?

### **1. How many finalists are there per class?**

Before getting the number of contestants per class, we first define the classes:

| Class | Meaning | Definition |
|:---:|---| --- |
| W | Winner | Contestant has won the season |
| R | Runner-Up | Contestant became the runner-up or placed 2nd |
| E | Elimenated | Contestant was elimenated at the finale and placed 3rd/4th |

With that, we now get the total of contestants by counting them in the `W/R/E` column.

In [142]:
df[["W/R/E"]].count()

W/R/E    160
dtype: int64

Now, we get the division of classes between all 160 contestants by printing the `.value_counts()` and summing it all up to measure.

In [143]:
print("Division of classes:", df[["W/R/E"]].value_counts())
print("Sum of the classes:", df[["W/R/E"]].value_counts().sum())

Division of classes: W/R/E
R        69
W        46
E        45
dtype: int64
Sum of the classes: 160


### **2. How many unique values are there in the `Season` column?**

Similar to the previous question, we first get the count of seasons available.

In [144]:
df[['Season']].count()

Season    160
dtype: int64

From getting the value counts in the `Season` column, we can see that there are inconsistensies in the dataset. Here, we can take note of how to format the dataset when it comes to preprocessing the dataframe.

In [145]:
df[['Season']].value_counts()

Season
1         26
S1        24
2         16
S2        10
3          7
14         5
15         4
AS6        4
S13        4
S11        4
AS7        4
S10        4
AS4        4
AS3        4
AS2        4
AS1        4
4          4
S9         4
AS5        3
S12        3
S3         3
S4         3
S5         3
S6         3
S7         3
S8         3
dtype: int64

### **3. How many unique values are there in the `Country` column?**

In [146]:
df[["Country"]].count()

Country    160
dtype: int64

By seeing the value counts in the `Country` column, we can observer that there are multiple values. However, countries with `_WORLD` just classify them as a different country rather then a *type* of season. When in fact, `_WORLD` seasons are similar to an All Star season.

In [147]:
df[["Country"]].value_counts()

Country  
US           79
UK           14
CAN          10
ESP           8
AUS           7
HOL           7
ITA           7
THI           6
CAN_WORLD     4
FRA           4
PH            4
UK_WORLD      4
BEL           3
SWE           3
dtype: int64

### **4. What are the measure of central tendency in the `Episode` column?**

First, we check the data type in the `Episode` column.

In [148]:
df['Episode'].dtypes

dtype('float64')

Since the values in the episode column are floats, we can `describe` the column to get the inferential statistics.

In [149]:
df["Episode"].describe()

count    160.00000
mean       8.81875
std        2.40511
min        5.00000
25%        7.00000
50%        9.00000
75%       11.00000
max       13.00000
Name: Episode, dtype: float64

In [150]:
print("Max:", df['Episode'].max()),
print("Min:", df['Episode'].min()),
print("Range:", df['Episode'].max() - df['Episode'].min())

Max: 13.0
Min: 5.0
Range: 8.0


## **Data Preprocessing**

#### **Data Preprocessing**

##### **Dropping Unneeded Columns**

In [151]:
df.head()

Rank WRE   Rank Country Season    Code W/R/E          Queen  Z-Score  \
0       NaN    NaN     NaN    NaN     NaN   NaN            NaN      NaN   
1      18.0    8.0      US    AS1  US_AS1     W  Chad Michaels     0.47   
2     131.0  159.0      US    AS1  US_AS1     R          Raven    -2.26   
3      77.0  159.0      US    AS1  US_AS1     E        Jujubee    -2.26   
4       6.0    8.0      US    AS1  US_AS1     E        Shannel     0.47   

    Percent  Score  ...  Low  Safe  High  SWin  Win  LSA    T  Unnamed: 36  \
0       NaN    NaN  ...   -1     0     1   1.5    2  NaN  NaN     Never/--   
1   100.00%    5.0  ...    0     0     1   0.0    3  NaN  NaN           --   
2  -120.00%   -6.0  ...    1     0     1   0.0    0  LSA  2.0           --   
3  -120.00%   -6.0  ...    1     0     1   0.0    0  LSA  2.0           --   
4   100.00%    5.0  ...    0     0     1   0.0    3  NaN  NaN           --   

   Unnamed: 37  Unnamed: 38  
0         Rank         Rank  
1        1,220            1  
2          237          140  
3          237          140  
4        1,220            1  

[5 rows x 39 columns]

In [152]:
df.columns

Index(['Rank WRE', 'Rank', 'Country', 'Season', 'Code', 'W/R/E', 'Queen',
       'Z-Score', 'Percent', 'Score', 'Episode', '1', '2', '3', '4', '5', '6',
       '7', '8', '9', '10', '11', '12', '13', '14', '15', 'Mean: 3.11',
       'Unnamed: 27', 'Bttm', 'Low', 'Safe', 'High', 'SWin', 'Win', 'LSA', 'T',
       'Unnamed: 36', 'Unnamed: 37', 'Unnamed: 38'],
      dtype='object')

In [153]:
drop_col = ['Rank WRE', 'Rank', 'Code', 'Episode', 'Percent', 'Score', 
            'Mean: 3.11', 'Z-Score', 'Unnamed: 27', 'LSA', 'T', 
            'Unnamed: 36', 'Unnamed: 37', 'Unnamed: 38']

In [154]:
df = df.drop(drop_col, axis=1) #Dropping the columns
df = df.iloc[1:,:] #Dropping the first row

In [155]:
df.head() #checking the dataset

Country Season W/R/E          Queen    1    2    3    4    5    6  ...  12  \
1      US    AS1     W  Chad Michaels -2.0  1.0  2.0  2.0  2.0  NaN  ... NaN   
2      US    AS1     R          Raven  1.0 -1.0 -2.0 -2.0 -2.0  NaN  ... NaN   
3      US    AS1     E        Jujubee  1.0 -1.0 -2.0 -2.0 -2.0  NaN  ... NaN   
4      US    AS1     E        Shannel -2.0  1.0  2.0  2.0  2.0  NaN  ... NaN   
5      US    AS2     W         Alaska  1.0  2.0  0.0  2.0  2.0  2.0  ... NaN   

   13  14  15  Bttm  Low  Safe  High  SWin  Win  
1 NaN NaN NaN     1    0     0     1   0.0    3  
2 NaN NaN NaN     3    1     0     1   0.0    0  
3 NaN NaN NaN     3    1     0     1   0.0    0  
4 NaN NaN NaN     1    0     0     1   0.0    3  
5 NaN NaN NaN     1    0     1     1   0.0    4  

[5 rows x 25 columns]

##### **Regex Functions**

In [156]:
#regex function to remove S but not remove the S in AS
def remove_s(x):
    return re.sub(r'S', '', x)

#regex function to remove AS and _WORLD
def remove_as(x):
    return re.sub(r'AS|_WORLD', '', x)

#regex function to remove the letter A
def remove_a(x):
    return re.sub(r'A', '', x)


##### **Removing `S` from the Season column**

In [157]:
df['Season'] = df['Season'].apply(remove_s)
df['Season'].value_counts()

1     50
2     26
3     10
4      7
14     5
A7     4
A6     4
9      4
A2     4
A1     4
13     4
11     4
10     4
A4     4
A3     4
15     4
12     3
5      3
6      3
7      3
8      3
A5     3
Name: Season, dtype: int64

With this new column, the S has been removed and only the All Stars Seasons are identified

##### **Adding Type Column**

Here we identify the type of season if it was a Regular, All-Star, or World season.

In [158]:
df.head()

Country Season W/R/E          Queen    1    2    3    4    5    6  ...  12  \
1      US     A1     W  Chad Michaels -2.0  1.0  2.0  2.0  2.0  NaN  ... NaN   
2      US     A1     R          Raven  1.0 -1.0 -2.0 -2.0 -2.0  NaN  ... NaN   
3      US     A1     E        Jujubee  1.0 -1.0 -2.0 -2.0 -2.0  NaN  ... NaN   
4      US     A1     E        Shannel -2.0  1.0  2.0  2.0  2.0  NaN  ... NaN   
5      US     A2     W         Alaska  1.0  2.0  0.0  2.0  2.0  2.0  ... NaN   

   13  14  15  Bttm  Low  Safe  High  SWin  Win  
1 NaN NaN NaN     1    0     0     1   0.0    3  
2 NaN NaN NaN     3    1     0     1   0.0    0  
3 NaN NaN NaN     3    1     0     1   0.0    0  
4 NaN NaN NaN     1    0     0     1   0.0    3  
5 NaN NaN NaN     1    0     1     1   0.0    4  

[5 rows x 25 columns]

##### **Regex Functions for New Column**

In [169]:
def identify_world(x):
    if '_WORLD' in x:
        return 'WORLD',
    
def identify_as(x):    
    if 'A' in x:
        return 'AS'

def identify_reg(x):
    if 'A' not in x:
        return 'REG'
    
def remove_world_reg(df, column_name):
    df[column_name] = df[column_name].str.replace(r"\('WORLD',\)REG", "WORLD", regex=True)
    return df
    


##### **Extracting the formats**

In [160]:
df['world'] = df['Country'].apply(identify_world).astype(str)
df['world'] = df['world'].replace('None', '')
df['world'].value_counts()

              152
('WORLD',)      8
Name: world, dtype: int64

In [161]:
df['allstars'] = df['Season'].apply(identify_as).astype(str)
df['allstars'] = df['allstars'].replace('None', '')
df['allstars'].value_counts()

      133
AS     27
Name: allstars, dtype: int64

In [162]:
df['regular'] = df['Season'].apply(identify_reg).astype(str)
df['regular'] = df['regular'].replace('None', '')
df['regular'].value_counts()

REG    133
        27
Name: regular, dtype: int64

In [168]:
#joining two columns into one column
df['format'] = df['world'] + df['allstars'] + df['regular']

df['format'] = df['format'].replace(np.nan, 'REG')

df['format'].value_counts()

REG              125
AS                27
('WORLD',)REG      8
Name: format, dtype: int64

#### **Data Cleaning**

##### **Fixing the Columns**

In [164]:
df['Season'] = df['Season'].apply(remove_a)
df['Season'].value_counts()

1     54
2     30
3     14
4     11
6      7
7      7
5      6
14     5
10     4
11     4
13     4
9      4
15     4
12     3
8      3
Name: Season, dtype: int64

In [171]:
df = remove_world_reg(df, 'format')
df['format'].value_counts().sum()

160

##### **Removing Type in Season**

In [177]:
df['Country'] = df['Country'].apply(remove_as)
df['Country'].value_counts()

US     79
UK     18
CAN    14
ESP     8
HOL     7
AUS     7
ITA     7
THI     6
FRA     4
PH      4
BEL     3
SWE     3
Name: Country, dtype: int64

##### **Changing Placement Values**

In [178]:
df.columns

Index(['Country', 'Season', 'W/R/E', 'Queen', '1', '2', '3', '4', '5', '6',
       '7', '8', '9', '10', '11', '12', '13', '14', '15', 'Bttm', 'Low',
       'Safe', 'High', 'SWin', 'Win', 'world', 'allstars', 'regular',
       'format'],
      dtype='object')

In [197]:
episodes =[ '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14']

In [198]:
#replacing values with another value in multiple columns
df[episodes] = df[episodes].replace({
    -2 : 0, #Bottom 2 placement
    -1 : 1, #Received negative critiques, but was not up for elimination   
    0 : 2, #Safe 
    1 : 3, #Received positive critiques, but did not win the challenge 
    1.5 : 4, #Top 2 contestants (Applicable in All Stars and World seasons and selected episodes in regular seasons) but lost the lipsync
    2 : 5 #Won the main challenge
    })

C:\Users\Mos\AppData\Local\Temp\ipykernel_24344\4080025004.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df[episodes] = df[episodes].replace({


In [186]:
df[episodes].describe()

1           2           3           4           5           6  \
count  154.000000  154.000000  156.000000  160.000000  159.000000  142.000000   
mean     4.201299    4.038961    4.064103    3.965625    3.911950    3.774648   
std      1.012212    1.142656    1.134102    1.192379    1.192719    1.163233   
min      2.000000    2.000000    2.000000    0.500000    2.000000    2.000000   
25%      3.000000    3.000000    3.000000    3.000000    3.000000    3.000000   
50%      5.000000    5.000000    5.000000    5.000000    5.000000    3.000000   
75%      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

                7          8          9         10         11         12  \
count  139.000000  93.000000  89.000000  51.000000  54.000000  43.000000   
mean     3.647482   3.655914   3.685393   3.470588   3.722222   4.209302   
std      1.196997   1.193229   1.220986   1.155040   1.156061   1.186394   
min      2.000000   2.000000   2.000000   2.000000   2.000000   2.000000   
25%      3.000000   3.000000   3.000000   3.000000   3.000000   3.000000   
50%      3.000000   3.000000   3.000000   3.000000   3.000000   5.000000   
75%      5.000000   5.000000   5.000000   5.000000   5.000000   5.000000   
max      5.000000   5.000000   5.000000   5.000000   5.000000   5.000000   

              13        14   15  
count  13.000000  9.000000  0.0  
mean    3.307692  3.888889  NaN  
std     1.250641  1.364225  NaN  
min     2.000000  2.000000  NaN  
25%     2.000000  3.000000  NaN  
50%     3.000000  5.000000  NaN  
75%     5.000000  5.000000  NaN  
max     5.000000  5.000000  NaN

##### **Sum of placement scores**

In [199]:
#sum all numbers per row in columns 1-14
df['score'] = df[episodes].sum(axis=1)

df

C:\Users\Mos\AppData\Local\Temp\ipykernel_24344\533511160.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['score'] = df[episodes].sum(axis=1)


season country format w/r/e                queen    1    2    3    4    5  \
1        1      US     AS     W        Chad Michaels  5.0  3.0  5.0  5.0  5.0   
2        1      US     AS     R                Raven  3.0  3.0  5.0  5.0  5.0   
3        1      US     AS     E              Jujubee  3.0  3.0  5.0  5.0  5.0   
4        1      US     AS     E              Shannel  5.0  3.0  5.0  5.0  5.0   
5        2      US     AS     W               Alaska  3.0  5.0  5.0  5.0  5.0   
..     ...     ...    ...   ...                  ...  ...  ...  ...  ...  ...   
156      1     BEL    REG     W         Drag Couenne  3.0  5.0  5.0  5.0  5.0   
157      1     BEL    REG     E                Susan  5.0  3.0  5.0  5.0  3.0   
158      1     SWE    REG     W  Admira Thunderpussy  3.0  5.0  5.0  5.0  5.0   
159      1     SWE    REG     R              Fontana  5.0  5.0  3.0  5.0  5.0   
160      1     SWE    REG     E          Vanity Vain  5.0  5.0  5.0  3.0  3.0   

       6    7   8   9  10  11  12  13  14  score  
1    NaN  NaN NaN NaN NaN NaN NaN NaN NaN   23.0  
2    NaN  NaN NaN NaN NaN NaN NaN NaN NaN   21.0  
3    NaN  NaN NaN NaN NaN NaN NaN NaN NaN   21.0  
4    NaN  NaN NaN NaN NaN NaN NaN NaN NaN   23.0  
5    5.0  5.0 NaN NaN NaN NaN NaN NaN NaN   33.0  
..   ...  ...  ..  ..  ..  ..  ..  ..  ..    ...  
156  3.0  5.0 NaN NaN NaN NaN NaN NaN NaN   31.0  
157  3.0  3.0 NaN NaN NaN NaN NaN NaN NaN   27.0  
158  3.0  5.0 NaN NaN NaN NaN NaN NaN NaN   31.0  
159  5.0  5.0 NaN NaN NaN NaN NaN NaN NaN   33.0  
160  3.0  5.0 NaN NaN NaN NaN NaN NaN NaN   29.0  

[160 rows x 20 columns]

##### **Final Column Fixes**

In [189]:
df.columns

Index(['Country', 'Season', 'W/R/E', 'Queen', '1', '2', '3', '4', '5', '6',
       '7', '8', '9', '10', '11', '12', '13', '14', 'Bttm', 'Low', 'Safe',
       'High', 'SWin', 'Win', 'format'],
      dtype='object')

In [188]:
drop_col = ['15', 'world', 'allstars', 'regular']
df = df.drop(drop_col, axis=1)
df

Country Season W/R/E                Queen    1    2    3    4    5    6  \
1        US      1     W        Chad Michaels  2.0  3.0  5.0  5.0  5.0  NaN   
2        US      1     R                Raven  3.0  3.0  2.0  2.0  2.0  NaN   
3        US      1     E              Jujubee  3.0  3.0  2.0  2.0  2.0  NaN   
4        US      1     E              Shannel  2.0  3.0  5.0  5.0  5.0  NaN   
5        US      2     W               Alaska  3.0  5.0  5.0  5.0  5.0  5.0   
..      ...    ...   ...                  ...  ...  ...  ...  ...  ...  ...   
156     BEL      1     W         Drag Couenne  3.0  5.0  5.0  5.0  5.0  3.0   
157     BEL      1     E                Susan  5.0  3.0  5.0  2.0  3.0  3.0   
158     SWE      1     W  Admira Thunderpussy  3.0  5.0  5.0  5.0  5.0  3.0   
159     SWE      1     R              Fontana  5.0  5.0  3.0  5.0  2.0  5.0   
160     SWE      1     E          Vanity Vain  5.0  2.0  5.0  3.0  3.0  3.0   

     ...  12  13  14  Bttm  Low  Safe  High  SWin  Win  format  
1    ... NaN NaN NaN     1    0     0     1   0.0    3      AS  
2    ... NaN NaN NaN     3    1     0     1   0.0    0      AS  
3    ... NaN NaN NaN     3    1     0     1   0.0    0      AS  
4    ... NaN NaN NaN     1    0     0     1   0.0    3      AS  
5    ... NaN NaN NaN     1    0     1     1   0.0    4      AS  
..   ...  ..  ..  ..   ...  ...   ...   ...   ...  ...     ...  
156  ... NaN NaN NaN     0    0     2     2   0.0    3     REG  
157  ... NaN NaN NaN     1    3     1     1   0.0    1     REG  
158  ... NaN NaN NaN     0    1     2     1   0.0    3     REG  
159  ... NaN NaN NaN     1    1     4     0   0.0    1     REG  
160  ... NaN NaN NaN     2    1     1     2   0.0    1     REG  

[160 rows x 25 columns]

In [190]:
#lowercase all column names
df.columns = map(str.lower, df.columns)

#reorder pandas columns
df = df[['season', 'country', 'format', 'w/r/e', 'queen', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14']]

season country format w/r/e                queen    1    2    3    4    5  \
1        1      US     AS     W        Chad Michaels  2.0  3.0  5.0  5.0  5.0   
2        1      US     AS     R                Raven  3.0  3.0  2.0  2.0  2.0   
3        1      US     AS     E              Jujubee  3.0  3.0  2.0  2.0  2.0   
4        1      US     AS     E              Shannel  2.0  3.0  5.0  5.0  5.0   
5        2      US     AS     W               Alaska  3.0  5.0  5.0  5.0  5.0   
..     ...     ...    ...   ...                  ...  ...  ...  ...  ...  ...   
156      1     BEL    REG     W         Drag Couenne  3.0  5.0  5.0  5.0  5.0   
157      1     BEL    REG     E                Susan  5.0  3.0  5.0  2.0  3.0   
158      1     SWE    REG     W  Admira Thunderpussy  3.0  5.0  5.0  5.0  5.0   
159      1     SWE    REG     R              Fontana  5.0  5.0  3.0  5.0  2.0   
160      1     SWE    REG     E          Vanity Vain  5.0  2.0  5.0  3.0  3.0   

       6    7   8   9  10  11  12  13  14  
1    NaN  NaN NaN NaN NaN NaN NaN NaN NaN  
2    NaN  NaN NaN NaN NaN NaN NaN NaN NaN  
3    NaN  NaN NaN NaN NaN NaN NaN NaN NaN  
4    NaN  NaN NaN NaN NaN NaN NaN NaN NaN  
5    5.0  2.0 NaN NaN NaN NaN NaN NaN NaN  
..   ...  ...  ..  ..  ..  ..  ..  ..  ..  
156  3.0  5.0 NaN NaN NaN NaN NaN NaN NaN  
157  3.0  3.0 NaN NaN NaN NaN NaN NaN NaN  
158  3.0  5.0 NaN NaN NaN NaN NaN NaN NaN  
159  5.0  5.0 NaN NaN NaN NaN NaN NaN NaN  
160  3.0  2.0 NaN NaN NaN NaN NaN NaN NaN  

[160 rows x 19 columns]

## **Exploratory Data Analysis Part 2**

The following questions are asked to guide the EDA.

1. 

In [195]:
df

season country format w/r/e                queen    1    2    3    4    5  \
1        1      US     AS     W        Chad Michaels  2.0  3.0  5.0  5.0  5.0   
2        1      US     AS     R                Raven  3.0  3.0  2.0  2.0  2.0   
3        1      US     AS     E              Jujubee  3.0  3.0  2.0  2.0  2.0   
4        1      US     AS     E              Shannel  2.0  3.0  5.0  5.0  5.0   
5        2      US     AS     W               Alaska  3.0  5.0  5.0  5.0  5.0   
..     ...     ...    ...   ...                  ...  ...  ...  ...  ...  ...   
156      1     BEL    REG     W         Drag Couenne  3.0  5.0  5.0  5.0  5.0   
157      1     BEL    REG     E                Susan  5.0  3.0  5.0  2.0  3.0   
158      1     SWE    REG     W  Admira Thunderpussy  3.0  5.0  5.0  5.0  5.0   
159      1     SWE    REG     R              Fontana  5.0  5.0  3.0  5.0  2.0   
160      1     SWE    REG     E          Vanity Vain  5.0  2.0  5.0  3.0  3.0   

       6    7   8   9  10  11  12  13  14  
1    NaN  NaN NaN NaN NaN NaN NaN NaN NaN  
2    NaN  NaN NaN NaN NaN NaN NaN NaN NaN  
3    NaN  NaN NaN NaN NaN NaN NaN NaN NaN  
4    NaN  NaN NaN NaN NaN NaN NaN NaN NaN  
5    5.0  2.0 NaN NaN NaN NaN NaN NaN NaN  
..   ...  ...  ..  ..  ..  ..  ..  ..  ..  
156  3.0  5.0 NaN NaN NaN NaN NaN NaN NaN  
157  3.0  3.0 NaN NaN NaN NaN NaN NaN NaN  
158  3.0  5.0 NaN NaN NaN NaN NaN NaN NaN  
159  5.0  5.0 NaN NaN NaN NaN NaN NaN NaN  
160  3.0  2.0 NaN NaN NaN NaN NaN NaN NaN  

[160 rows x 19 columns]

# **Saving Dataframes as CSVs**

In [166]:
#main_df.to_csv('data\main.csv')
